### PR TITLE
Performed miscellaneous multi deployer cleanups.

### PR DESCRIPTION
**Pull request by mwhittaker**
This PR cleans up the weaver multi deployer.

- I removed the use of `protos.Deployment`. I'm hoping we can remove this proto eventually. It should be obsolete now.
- The deployer now instantiates its own logs and trace databases.
- I renamed some variables and updates some stale comments.

### CODE REVIEW

File: internal/tool/multi/deployer.go [hunk 1]
```diff
@@ -405,16 +394,13 @@ func (d *deployer) registerReplica(g *group, info *protos.WeaveletInfo) error {
...
 }
 
 // HandleTraceSpans implements the envelope.EnvelopeHandler interface.
-func (d *deployer) HandleTraceSpans(_ context.Context, spans []trace.ReadOnlySpan) error {
-	if d.traceSaver == nil {
-		return nil
-	}
-	return d.traceSaver(spans)
+func (d *deployer) HandleTraceSpans(ctx context.Context, spans []trace.ReadOnlySpan) error {
+	return d.traceDB.Store(ctx, d.config.Name, d.deploymentId, spans)
```
Comment (rgrandl) [inline internal/tool/multi/deployer.go:403]: ditto

File: internal/tool/multi/deployer.go [hunk 2]
```diff
@@ -405,16 +394,13 @@ func (d *deployer) registerReplica(g *group, info *protos.WeaveletInfo) error {
 
 // HandleLogEntry implements the envelope.EnvelopeHandler interface.
 func (d *deployer) HandleLogEntry(_ context.Context, entry *protos.LogEntry) error {
-	d.logSaver(entry)
+	d.logsDB.Add(entry)
```
Comment (rgrandl) [inline internal/tool/multi/deployer.go:397]: wouldn't this panic if logsDB is nil?

Somehow it assumes that someone created a deployer with newDeployer.

File: internal/tool/multi/deployer.go [hunk 3]
```diff
@@ -48,27 +48,15 @@ const defaultReplication = 2
...
 // A deployer manages an application deployment.
 type deployer struct {
-	ctx       context.Context
-	ctxCancel context.CancelFunc
-	dep       *protos.Deployment
-	started   time.Time
-	logger    *slog.Logger
-	running   errgroup.Group
-
-	// logSaver processes log entries generated by the weavelet. The entries
```
Comment (rgrandl) [inline internal/tool/multi/deployer.go:58]: I like that we could delete these functions.

**Review by rgrandl — APPROVED** [review]
Nice cleanups Michael!

File: internal/tool/multi/deployer.go (hunk 3 above)
Comment (mwhittaker) [reply internal/tool/multi/deployer.go:58]: Me too! I think when the multiprocess deployer was used for weavertests, we had to pass these in, but now we don't :)

File: internal/tool/multi/deployer.go (hunk 2 above)
Comment (mwhittaker) [reply internal/tool/multi/deployer.go:397]: That's true! But, I don't think any of a deployer's methods would work if you don't construct it with `newDeployer`, so I think it's okay?

**Review by mwhittaker — COMMENTED** [review]
Thanks Robert!

**Pull request merged**